src/scatter_buildblock/ScatterEstimation.cxx

```diff
@@ -0,0 +1,1486 @@
+            pow_times_add add_scalar (-1e-9f, 1.0f, 1.0f, NumericInfo<float>().min_value(), NumericInfo<float>().max_value());
+
+            apply_to_proj_data(*temp_projdata, min_threshold);
+            apply_to_proj_data(*temp_projdata, add_scalar);
+
+            // ok, we can multiply with the norm
+            normalisation_factors_sptr->apply(*temp_projdata, start_time, end_time);
+
+            std::stringstream convert;   // stream used for the conversion
+            convert << this->output_scatter_estimate_prefix << "_" <<
+                       i_scat_iter;
+            std::string output_scatter_filename = convert.str();
+
+            // To save the 3d scatter estimate
+            shared_ptr <ProjData> temp_scatter_projdata_3d (
+                        new ProjDataInterfile(this->input_projdata_sptr->get_exam_info_sptr(),
+                                              this->input_projdata_sptr->get_proj_data_info_sptr() ,
+                                              output_scatter_filename,
+                                              std::ios::in | std::ios::out | std::ios::trunc));
+            shared_ptr<BinNormalisation> dummy_normalisation_coeffs_3d_sptr(new TrivialBinNormalisation());
+            // Upsample to 3D
+            //we're currently not doing the tail fitting in this step, but keeping the same scale as determined in 2D
+
+            upsample_and_fit_scatter_estimate(*temp_scatter_projdata_3d,
+                                              *this->input_projdata_sptr,
+                                              *temp_projdata,
+                                              *dummy_normalisation_coeffs_3d_sptr,
+                                              *this->input_projdata_sptr,
+                                              1.0f, 1.0f, 1, spline_type,
+                                              false);
+
+            // Now save the full background term.
+            convert.clear();
+            convert << this->output_background_estimate_prefix << "_" <<
+                       i_scat_iter;
+            std::string output_background_filename = convert.str();
+
+            shared_ptr <ProjData> temp_background_projdata_3d (
+                        new ProjDataInterfile(this->input_projdata_sptr->get_exam_info_sptr(),
+                                              this->input_projdata_sptr->get_proj_data_info_sptr() ,
+                                              output_background_filename,
+                                              std::ios::in | std::ios::out | std::ios::trunc));
+            temp_background_projdata_3d->fill(*temp_scatter_projdata_3d);
+
+            if (!is_null_ptr(add_projdata_sptr))
+                add_proj_data(*temp_background_projdata_3d, *this->add_projdata_sptr);
+            multiplicative_binnorm_sptr->apply(*temp_background_projdata_3d, start_time, end_time);
```

Comment:
terminology is now confusing, as this `background` is now the `additive` term...

Reply:
Fixed